src/use-swr.ts

```diff
@@ -719,7 +719,7 @@ function useSWR<Data = any, Error = any>(
       isValidating: {
         get: function() {
           stateDependencies.current.isValidating = true
-          return stateRef.current.isValidating
+          return key ? stateRef.current.isValidating : false
```

Comment:
I can only come up with this simple solution that return `false` when key is falsy.

since when key changes swr will always clean up and `unmountedRef.current` changes to `true`. so dispatch won't work and in won't pass the internal condition.

Reply:
This fix sounds good until we have a better and complete refactoring to the codebase. 👍